endpoints/openrtb2/auction.go

```diff
@@ -1633,3 +1642,48 @@ func checkIfAppRequest(request []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+func (deps *endpointDeps) ExtractFPDForBidders(req *openrtb_ext.RequestWrapper) (map[openrtb_ext.BidderName]*openrtb_ext.ORTB2, []error) {
+	errL := []error{}
+	var resolvedFPD map[openrtb_ext.BidderName]*openrtb_ext.ORTB2
+
+	//If {site,app,user}.ext.data exists, collect it and remove {site,app,user}.ext.data from request
+	globalFpdData, err := firstpartydata.ExtractGlobalFPDData(req)
+	if err != nil {
+		errL = []error{err}
+		return resolvedFPD, errL
+	}
+
+	reqExt, err := req.GetRequestExt()
+	if err != nil {
+		errL = append(errL, err)
+		return resolvedFPD, errL
+	}
+	if reqExt == nil || reqExt.GetPrebid() == nil {
+		return resolvedFPD, errL
+	}
+	biddersWithGlobalFPD := make([]string, 0)
+
+	if reqExt.GetPrebid().Data != nil {
+		biddersWithGlobalFPD = reqExt.GetPrebid().Data.Bidders
+		reqExt.GetPrebid().Data.Bidders = nil
+	}
```

Comment:
`reqExt.GetPrebid` allocates a shallow copy. Please assign it once and then reference the variable. Something like:

```
if prebidData := reqExt.GetPrebid().Data; prebidData != nil {
	biddersWithGlobalFPD = prebidData.Bidders
	prebidData.Bidders = nil
}
```

Reply:
Modified